### PR TITLE
fix(server): Rate limit outcomes emited only for events on "fast-path"

### PR DESCRIPTION
**Pull request by RaduW**
Fixes outcome generation on the "fast-path" i.e. when we know from the cache that we are rate limited and 
we don't try to process the envelope.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 - Accept big-endian minidumps. ([#789](https://github.com/getsentry/relay/pull/789))
 - Detect network outages and retry sending events instead of dropping them. ([#788](https://github.com/getsentry/relay/pull/788))
 - Rate limit outcomes emitted only for events. ([#806](https://github.com/getsentry/relay/pull/806))
+- Rate limit outcomes emited only for events on "fast-path". ([#809](https://github.com/getsentry/relay/pull/809))
```
Comment (jan-auer) [inline CHANGELOG.md]: Can you merge the two changelog entries and move them to _Internal_?

File: relay-quotas/src/rate_limit.rs [hunk 2]
```diff
@@ -286,6 +286,21 @@ impl RateLimits {
     pub fn longest(&self) -> Option<&RateLimit> {
         self.iter().max_by_key(|limit| limit.retry_after)
     }
+
+    /// Returns the most relevant rate limit.
+    ///
+    /// The most relevant rate limit is the longest rate limit for events and if there is no
```
Comment (jan-auer) [inline relay-quotas/src/rate_limit.rs]: You could clarify in the doc string that this is for error events only.

File: relay-server/src/actors/events.rs [hunk 3]
```diff
@@ -1799,7 +1784,7 @@ mod tests {
         });
 
         // only non event rate limits so nothing relevant
-        assert_eq!(most_relevant(&rate_limits), None)
+        assert_eq!(rate_limits.longest_error(), None)
```
Comment (jan-auer) [inline relay-server/src/actors/events.rs]: Let's move this test to the module where the function is declared.

File: relay-server/src/endpoints/common.rs [hunk 4]
```diff
@@ -457,15 +455,16 @@ where
...
+                        timestamp: start_time,
+                        scoping: *scoping.borrow(),
+                        outcome,
+                        event_id: *event_id.borrow(),
+                        remote_addr,
+                    });
+                }
             }
-
             if !emit_rate_limit && matches!(error, BadStoreRequest::RateLimited(_)) {
```
Comment (jan-auer) [inline relay-server/src/endpoints/common.rs:469]: Looks like this empty line was removed unintentionally.

File: CHANGELOG.md (hunk 1 above)
Comment (RaduW) [reply CHANGELOG.md]: ok

File: relay-quotas/src/rate_limit.rs (hunk 2 above)
Comment (RaduW) [reply relay-quotas/src/rate_limit.rs]: yep

**Review by jan-auer — APPROVED** [review]

**Pull request merged**